ambry-api/src/main/java/com/github/ambry/config/CompressionConfig.java

```diff
@@ -0,0 +1,319 @@
+ */
+public class CompressionConfig {
+
+  // Boolean whether compression is enabled in PUT operation.
+  private static final String CONFIG_COMPRESSION_ENABLED = "router.compression.enabled";
+  private static final boolean DEFAULT_COMPRESSION_ENABLED = true;
+
+  // Whether to skip compression if content-encoding present.
+  private static final String CONFIG_SKIP_CONTENT_ENCODING = "router.compression.skip.content.encoding";
+  private static final boolean DEFAULT_SKIP_CONTENT_ENCODING = true;
+
+  // A comma-separated list of compressible fully content-types, like "text/javascript, text/log; Charset=UTF8"
+  private static final String CONFIG_CONTENT_TYPES_COMPRESSIBLE = "router.compression.content.types.compressible";
+  private static final String DEFAULT_CONTENT_TYPES_COMPRESSIBLE = null;
+
+  // A comma-separated list of incompressible fully content-types, like "text/javascript, text/log; Charset=UTF8"
+  private static final String CONFIG_CONTENT_TYPES_INCOMPRESSIBLE = "router.compression.content.types.incompressible";
+  private static final String DEFAULT_CONTENT_TYPES_INCOMPRESSIBLE = null;
+
+  // A comma-separated list of compressible content-prefixes, such as "text" for text/*
+  private static final String CONFIG_CONTENT_PREFIXES_COMPRESSIBLE = "router.compression.content.prefixes.compressible";
+  private static final String DEFAULT_CONTENT_PREFIXES_COMPRESSIBLE = "text";
+
+  // A comma-separated list of incompressible content-prefixes, such as "image" for image/*.
+  private static final String CONFIG_CONTENT_PREFIXES_INCOMPRESSIBLE = "router.compression.content.prefixes.incompressible";
+  private static final String DEFAULT_CONTENT_PREFIXES_INCOMPRESSIBLE = "image";
+
+  // Boolean whether to compress other content-types not specified in content-types and content-prefixes.
+  // It applies to PUT operation only.
+  private static final String CONFIG_COMPRESS_OTHER_CONTENT_TYPE = "router.compression.other.content.types";
+  private static final boolean DEFAULT_COMPRESS_OTHER_CONTENT_TYPE = false;
+
+  // The minimal source/chunk size in bytes in order to qualify for compression.
+  private static final String CONFIG_MINIMAL_DATA_SIZE_IN_BYTES = "router.compression.minimal.content.size";
+  private static final int DEFAULT_MINIMAL_DATA_SIZE_IN_BYTES = 1024;
+
+  // The minimal compression ratio (uncompressedSize/compressSize) in order to keep compressed content.
+  // The higher the compression ratio the better with 1 means the same size.
+  private static final String CONFIG_MINIMAL_COMPRESS_RATIO = "router.compression.minimal.ratio";
+  private static final float DEFAULT_MINIMAL_COMPRESS_RATIO = 1.1f;
+
+  // The name of the algorithm to use.  This name must be pre-registered in CompressionMap class.
+  private static final String CONFIG_ALGORITHM_NAME = "router.compression.algorithm.name";
+  private static final String DEFAULT_ALGORITHM_NAME = null;
```

Comment:
let's put some default algorithm here.

Reply:
Sure. Since the name constant is in common and can't access from here.  I'd have to put a string value here.